pkg/maps/policymap/policymap.go

```diff
@@ -219,7 +279,24 @@ func (pm *PolicyMap) AllowKey(k PolicyKey, proxyPort uint16) error {
 // protocol `proto`. It is assumed that `dport` and `proxyPort` are in host byte-order.
 func (pm *PolicyMap) Allow(id uint32, dport uint16, proto u8proto.U8proto, trafficDirection trafficdirection.TrafficDirection, proxyPort uint16) error {
 	key := newKey(id, dport, proto, trafficDirection)
-	entry := newEntry(proxyPort)
+	pef := NewPolicyEntryFlag(&PolicyEntryFlagParam{})
+	entry := newEntry(proxyPort, pef)
+	return pm.Update(&key, &entry)
+}
+
+// DenyKey pushes an entry into the PolicyMap for the given PolicyKey k.
+// Returns an error if the update of the PolicyMap fails.
+func (pm *PolicyMap) DenyKey(k PolicyKey) error {
```

Comment:
>I'm not sure if it's a holdover from older code, but it's confusing to see mentions in packages like this of DenyKey when the key has no allow/deny portion to it. We should clean this up so that the usage is clear in the usage of the entry to hold the deny flag.

Can you clarify this a bit? This `DenyKey` is a function wrapper that invokes the entry with the deny set.